src/operators/Isovolume/avtIsovolumeFilter.C

```diff
@@ -518,4 +468,223 @@ avtIsovolumeFilter::ModifyContract(avtContract_p in_spec)
+    if(clips[0] && clips[1])
+    {
+        vtkDataSet *intermediate = NULL;
+        intermediate = ExecuteSingleClip(in_ds, atts.GetLbound(), true);
+        if(intermediate->GetNumberOfCells() > 0)
+        {
+            out_ds = ExecuteSingleClip(intermediate, atts.GetUbound(), false);
+            intermediate->Delete();
+        }
+        else
+            out_ds = intermediate;
+    }
+    else if(clips[0])
+        out_ds = ExecuteSingleClip(in_ds, atts.GetLbound(), true);
+    else if(clips[1])
+        out_ds = ExecuteSingleClip(in_ds, atts.GetUbound(), false);
+    bool own = out_ds != in_ds;
+
+    //
+    // Make sure there's something there
+    //
+    if (out_ds->GetNumberOfCells() <= 0)
+    {
+        // We can't delete the out_ds unless we made it and own it.
+        // Otherwise, we're deleting the reference out from under the in_ds.
+        if(own)
+            out_ds->Delete();
+        return NULL;
+    }
+
+    //
+    // If we had poly data input, we want poly data output.  The VTK filter
+    // only returns unstructured grids, so convert that now.  Note: we don't
+    // necessarily have a ugrid, since it might be that we didn't process the
+    // dataset.
+    //
+    if (in_ds->GetDataObjectType() == VTK_POLY_DATA &&
+        out_ds->GetDataObjectType() == VTK_UNSTRUCTURED_GRID)
+    {
+        vtkUnstructuredGrid *ugrid = (vtkUnstructuredGrid *) out_ds;
+        vtkPolyData *out_pd = vtkPolyData::New();
+        out_pd->SetPoints(ugrid->GetPoints());
+        out_pd->GetPointData()->ShallowCopy(ugrid->GetPointData());
+        out_pd->GetCellData()->ShallowCopy(ugrid->GetCellData());
+        vtkIdType ncells = ugrid->GetNumberOfCells();
+        out_pd->Allocate(ncells);
+        for (vtkIdType i = 0 ; i < ncells ; i++)
+        {
+            int celltype = ugrid->GetCellType(i);
+            vtkIdType *pts, npts;
+            ugrid->GetCellPoints(i, npts, pts);
+            out_pd->InsertNextCell(celltype, npts, pts);
+        }
+        if(own)
+            out_ds->Delete();
+        out_ds = out_pd;
+        own = true;
+    }
+
+    avtDataRepresentation *out_dr = NULL;
+    if (out_ds != NULL)
+        out_dr = new avtDataRepresentation(out_ds,
+            in_dr->GetDomain(), in_dr->GetLabel());
+
+    if (own && out_ds != NULL)
+        out_ds->Delete();
+
+    visitTimer->StopTimer(timerHandle, "avtIsovolumeFilter::ExecuteDataTree_VTK");
+    return out_dr;
+}
+
+avtDataRepresentation *
+avtIsovolumeFilter::ExecuteData_VTKM(avtDataRepresentation *in_dr, std::vector<double> bounds, std::vector<bool> clips)
+{
+#ifndef HAVE_LIBVTKH
+    return NULL;
+#else
+    int timerHandle = visitTimer->StartTimer();
+    vtkh::DataSet *in_ds = in_dr->GetDataVTKm();
+    if (!in_ds || in_ds->GetNumberOfDomains() != 1)
+        return NULL;
+
+    std::string isoVar(activeVariable != NULL ? activeVariable
+                                              : pipelineVariable);
+
+    //VTKm version.
```

Comment:
Any reason you have VTKm and VTKh versions?

Reply:
Nope. I was experimenting and forgot to remove it.. Removed now.